posts/redux-sagas-all-you-need-to-know.md

```diff
@@ -0,0 +1,107 @@
+# <img src="https://redux-saga.js.org/img/Redux-Saga-Logo.png" width="50"> Redux Sagas: All You Need to Know
+In the world of React web development, managing state is an essential task. Redux is a popular state management library used by many to store and manage the state of the application.
+
+However, when it comes to handling complex async logic, Redux on its own isn't always enough. This is where Redux Sagas come in, providing a solution to deal with async logic in a more declarative way.
+
+In this blog post, we'll take a deep dive into Redux Sagas, exploring what they are, why we would use them, their advantages over other libraries, and some example scenarios to help better understand their implementation.
+ 
+## What are Redux Sagas?
+Redux Sagas have gained popularity among developers lately as a middleware library used to handle complex side effects (e.g. async operations such as API calls, timers, and socket connections) in an elegant way by using ES6 generators instead of traditional callbacks or promises. This allows you to pause and resume the function at any point, making the flow of data more easily understood and predictable.
+
+It also helps to decouple the business logic of handling asynchronous operations and state management. Using Sagas can help you to write cleaner and more maintainable code. It can also provide better error handling and more advanced testing capabilities, as you can test the Sagas' logic directly.
+
+They are designed to be run in parallel with your Redux reducers, which means they don't block the main thread and can be used in conjunction with other Redux middleware (e.g., Redux Thunk).
+ 
+## Why use Redux Sagas?
+One of the most significant advantages of Redux Sagas over other middleware like is their handling of side effects natively. As we know, side effects are the operations that affect our application's state outside the current scope of a function.
```

Comment:
I'd re-write `other middleware like` to `like-minded middleware` or even just `other middlware`.